sparkbox/accessible-components
# feat: configure Prettier auto-formatting

### Description:
Setup and Configure Prettier auto-formatting

### Spec:
Designs: [Designs](DESIGN_URL)

See Story: [FSA22V2-230](https://sparkbox.atlassian.net/browse/FSA22V2-230?atlOrigin=eyJpIjoiMWFkOTdhYTQ0MjU4NGJhZjgzMTc3YTM4MjBmNjg2YzIiLCJwIjoiaiJ9)

### Validation:

Work Done:
- install Prettier.
- add .prettierrc.json file to auto fix formatting issues that match eslint rules on save.
- add .prettierignore file with content matching .gitignore per the recommendation of prettier documentation to ignore formatting of files that should not be formatted.

Notes:
- I used [these instructions](https://prettier.io/docs/en/install.html) for installing Prettier.
- I used [this resource](https://prettier.io/docs/en/install.html) to determine that I should add what is in the .gitignore into the .prettierignore
- I named the file .prettierrc.json because the documentation indicated that convention
- Dustin asked that I add a pre-commit hook to have Prettier run on stage files before they are committed. I have started research on this and plan to pair on it with someone Monday or Tuesday.


#### To Validate:

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
2. Pull down all related branches.
3. In terminal: `npm run dev`.
4. Check [localhost:3000](http://localhost:3000/)for changes.
5. Run `npm run lint`.
6. Run `npm run prepare` for husky pre-commit hook.
7. Run axe Devtools on affected pages.
8. Confirm all tests pass.



To test, it's best to add a change in the index.js file (like extra spaces), add it and then try to commit. The script should correct it before it can be committed.
---

#### Browser Testing


In these browsers, behavior & design closely match original specifications. A user is able to access all content and functionality, including the usability of required assistive devices, such as keyboard and screenreader.

**macOS**

* [ ] Safari (last 2 major versions)
* [ ] Chrome (last 6 months)
* [ ] Firefox (last 6 months)

**Windows**

* [ ] Chrome (last 6 months)
* [ ] Firefox (last 6 months)
* [ ] Edge (last 6 months)

**Mobile**

* [ ] Safari (last 2 major versions)
* [ ] Chrome on Android (last 6 months)
* [ ] Firefox on Android (last 6 months)
